rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -169,9 +210,22 @@ get_node_names(
     goto cleanup;
   }
 
+  if (security_contexts) {
+    rcutils_ret = rcutils_string_array_init(security_contexts, named_nodes_num, &allocator);
+
+    if (rcutils_ret != RCUTILS_RET_OK) {
+      RMW_SET_ERROR_MSG("could not allocate memory for node_namespaces output");
+      final_ret = rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+      goto cleanup;
+    }
+  }
+
   for (auto i = 0; i < named_nodes_num; ++i) {
     node_names->data[i] = rcutils_strdup(tmp_names_list.data[i], allocator);
     node_namespaces->data[i] = rcutils_strdup(tmp_namespaces_list.data[i], allocator);
+    if (security_contexts) {
+      security_contexts->data[i] = rcutils_strdup(tmp_security_contexts_list.data[i], allocator);
+    }
```

Comment:
@ivanpauno meta: this predates your PR, but why does this code assume that `rcutils_strdup` will succeed? Also, simply moving strings in `tmp` arrays would be much more efficient than this (de)allocation dance.

Reply:
I completely agree.
I will open an issue for this, because it's out of the scope of this PR IMO.